apps/native-component-list/src/screens/AV/VideoPlayer.tsx

```diff
@@ -127,7 +127,7 @@ function ResizeModeSegmentedControl({
 }: {
   onValueChange: (value: ResizeMode) => void;
 }) {
-  const resizeMap = {
+  const resizeMap: Record<string, ResizeMode | undefined> = {
```

Comment:
Why are you typing sth that is not representing an actual JS state? 🤔 
```suggestion
  const resizeMap: Record<string, ResizeMode> = {
```

Reply:
Because not for every `string` there's a value. To remove `undefined` I feel I should do `Record<'stretch' | 'contain' | 'cover', ResizeMode>` (as I said in some other command I may be abusing `Record`).

Reply:
Why do we need to specify the type for this variable like this in the first place? 🤔
Why not to go with implicit type inferred from the object decalration?
```ts
const resizeMap = {
  ...
};
```

Reply:
You can be more specific and do it like this:
```ts
const resizeMap = {
  ...
} as const
```

Reply:
Because in `onValueChange` callback the _new value_ is of type `string` for which there may be no mapping in the `resizeMap` map according to TS (it knows `resizeMap` only defines values for three specific literals):
<img width="1171" alt="Zrzut ekranu 2020-09-18 o 16 10 41" src="https://user-images.githubusercontent.com/1151041/93607358-81399a00-f9c9-11ea-9540-ae56eaf14c58.png">

Reply:
And this is perfectly logical, because `SegmentedControl` is given `values={data}` and `data = Object.keys(...) // string[]`. If `SegmentedControl` is properly inferring the type from the available values then it must be producing `string` type in `onChange` callback. I wonder if changing `values={data as (typeof keyof Screens)[]}` is not going to fix the issue with later casting 😉

Reply:
I know it is perfectly logical, I just post the error here for you to see exactly what I see. `SegmentedControl` does not infer types (if it would I would have used it). `values={data as (typeof keyof Screens)[]}` wouldn't change anything (we would still need to cast in `onChange`).

Reply:
Adding it would make sense if we wanted to have like, two places where we define that values in `values` and in `onChange` are expected and required to be `typeof keyof resizeMap`, but it is also (imho) error-prone (there are two places instead of one and we can't really guarantee on type level without `SegmentedControl` generics that casting in `onChange` will not be an error.

All in all I love types too, but I wouldn't say that omitting one `if (resizeMap[value])` is worth the inconveniences that trying to make this safe on type level which is currently impossible/very hard to do properly (given this 3rd party library type definitions).